lib/appoptics_apm/sdk/current_trace.rb

```diff
@@ -0,0 +1,69 @@
+#--
+# Copyright (c) 2019 SolarWinds, LLC.
+# All rights reserved.
+#++
+
+module AppOpticsAPM
+  module SDK
+
+    module CurrentTrace
+
+      # Creates an instance of {TraceId} with instance methods {TraceId#id} and {TraceId#for_log}.
+      #
+      # === Example:
+      #
+      #   trace = AppOpticsAPM::SDK.current_trace.new
+      #   trace.id        # '7435A9FE510AE4533414D425DADF4E180D2B4E36-0'
+      #   trace.for_log   # 'traceId=7435A9FE510AE4533414D425DADF4E180D2B4E36-0' or '' depends on Config
+      #
+      def current_trace
+        TraceId.new
+      end
+
+      # @attr id the current traceId, it looks like: '7435A9FE510AE4533414D425DADF4E180D2B4E36-0'
+      #          and ends in '-1' if the request is sampled and '-0' otherwise.
+      #          Results in '0000000000000000000000000000000000000000-0'
+      #          if the CurrentTrace instance was created outside of the context
+      #          of a request.
+      class TraceId
+        attr_reader :id
+
+        def initialize
+          @xtrace = AppOpticsAPM::Context.toString
+          task_id = AppOpticsAPM::XTrace.task_id(@xtrace)
+          sampled = AppOpticsAPM::XTrace.sampled?(@xtrace)
+          @id = "#{task_id}-#{sampled ? 1 : 0}"
+        end
+
+        # for_log returns a string in the format 'traceId=<current_trace.id>' or ''.
+        #          An empty string is returned depending on the setting for
+        #          <tt>AppOpticsAPM::Config[:log_traceId]</tt>, which can be :never,
```

Comment:
node's values are false, true, 'traced', 'sampled', 'always'. does't seem important but i thought i'd note it

Reply:
true == always ?
I wanted to avoid that because I don't think `always` should be a popular setting, I aimed for 4 words that described the setting (to some degree).